libflux/flux-core/src/semantic/doc.rs

```diff
@@ -559,6 +561,34 @@ fn comments_to_string(comments: &[ast::Comment]) -> String {
     s
 }
 
+/// Shorten removes all long form descriptions from the docs structures leaving only the headlines
+/// and other metadata.
+pub fn shorten(doc: &mut PackageDoc) {
+    doc.description = None;
+    for (_, m) in doc.members.iter_mut() {
+        remove_desc(m);
+    }
+}
+
+// removes the description from a Doc
+fn remove_desc(doc: &mut Doc) {
```

Comment:
I'm not sure that we need this function, as it's relatively short, and only called once, from the function right above it. To read this code, it would require pushing to the mental stack, and I'm not sure it provides much value for that. Thoughts?

Reply:
The function is recursive  with the `shorten` function. Its a bit odd to have two function that together make it recursive but this keeps the code much simpler than trying to convert the types into something that can be recursive with a single function. In short the Doc enum owns its data but we need a `&mut` to the data so that we don't have to make copies of it just to remove data.
 
 I'll add a comment to this effect.